crates/server/src/server/api/requests/view_file.rs

```diff
@@ -0,0 +1,140 @@
+use air_r_formatter::format_node;
+use air_r_parser::RParserOptions;
+use lsp_types::request::Request;
+use serde::Deserialize;
+use serde::Serialize;
+
+use crate::server::api::LSPResult;
+use crate::server::client::Notifier;
+use crate::server::Result;
+use crate::session::DocumentSnapshot;
+
+#[derive(Debug, PartialEq, Clone, Deserialize, Serialize)]
+pub(crate) enum ViewFileKind {
+    TreeSitter,
+    SyntaxTree,
+    FormatTree,
+}
+
+#[derive(Debug, PartialEq, Clone, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub(crate) struct ViewFileParams {
+    /// From `lsp_types::TextDocumentPositionParams`
+    pub(crate) text_document: lsp_types::TextDocumentIdentifier,
+    pub(crate) position: lsp_types::Position,
+
+    /// Viewer type
+    pub(crate) kind: ViewFileKind,
+}
+
+#[derive(Debug)]
+pub(crate) enum ViewFile {}
+
+impl Request for ViewFile {
+    type Params = ViewFileParams;
+    type Result = String;
+    const METHOD: &'static str = "air/viewFile";
+}
+
+impl super::RequestHandler for ViewFile {
+    type RequestType = ViewFile;
+}
+
+impl super::BackgroundDocumentRequestHandler for ViewFile {
```

Comment:
This is what implementing a custom request looks like. We have to implement `lsp_types::request::Request` for our custom request and supply the params/result types, but it's pretty easy!